### PR TITLE
QA-264: Delete pr_x branch in Gitlab upon a close or merge

### PR DESCRIPTION
**Pull request by oleorhagen**
Changelog: None
Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

File: menderqa_pipeline.go [hunk 1]
```diff
@@ -351,3 +355,57 @@ func stopBuildsOfStalePRs(log *logrus.Entry, pr *github.PullRequestEvent, conf *
 	return nil
 
 }
+
+func deleteStaleGitlabPRBranch(log *logrus.Entry, pr *github.PullRequestEvent, conf *config) error {
```
Comment (lluiscampos) [inline menderqa_pipeline.go:359]: This has nothing to do with `menderqa_pipeline`, it should be moved to `pullreq_pipeline`

File: main.go [hunk 2]
```diff
@@ -263,6 +263,11 @@ func processGitHubWebhook(ctx *gin.Context, payload []byte, githubClient *github
 			log.Errorf("Failed to stop a stale build after the PR: %v was merged or closed. Error: %v", pr, err)
 		}
 
+		// Delete merged pr branches in GitLab
+		if err = deleteStaleGitlabPRBranch(log, pr, conf); err != nil {
+			log.Errorf("Failed to delete the stale PR branch after the PR: %v was merged or closed. Error: %v", pr, err)
+		}
+
```
Comment (lluiscampos) [inline main.go:270]: Similar to my previous comment, this should be done before line 248, as from there on the processing only continues for `mendersoftware` members (iow mender-qa pipeline)

File: menderqa_pipeline.go [hunk 3]
```diff
@@ -351,3 +355,57 @@ func stopBuildsOfStalePRs(log *logrus.Entry, pr *github.PullRequestEvent, conf *
...
+
+	// If the action is "closed" the pull request was merged or just closed,
+	// stop builds in both cases.
+	if pr.GetAction() != "closed" {
+		log.Debugf("stopBuildsOfStalePRs: PR not closed, therefore not stopping it's pipeline")
```
Comment (lluiscampos) [inline menderqa_pipeline.go:364]: Typo in the function name

**Review by lluiscampos — CHANGES_REQUESTED** [review]
Looks good otherwise.

File: pullreq_pipeline.go [hunk 4]
```diff
@@ -68,3 +69,57 @@ func createPullRequestBranch(log *logrus.Entry, org, repo, pr, action string) er
 	log.Info("Pipeline is expected to start automatically")
 	return nil
 }
+
+func deleteStaleGitlabPRBranch(log *logrus.Entry, pr *github.PullRequestEvent, conf *config) error {
```
Comment (lluiscampos) [inline pullreq_pipeline.go:73]: For consistency, consider the same interface as the function above

```suggestion
func deleteStaleGitlabPRBranch(log *logrus.Entry, org, repo, pr, action string) error {
```

**Review by lluiscampos — APPROVED** [review]
Minor suggestion, change only if you wish.

File: pullreq_pipeline.go (hunk 4 above)
Comment (oleorhagen) [reply pullreq_pipeline.go:73]: I agreed, looking at it, but from looking in the main code, I actually think this is the saner interface, so leaving for now :)

**Pull request merged**